src/parser.rs

```diff
@@ -324,6 +371,48 @@ pub fn parse_gcov(gcov_path: &Path) -> Result<Vec<(String, CovResult)>, ParserEr
         .unwrap_or_else(|_| panic!("Failed to open gcov file {}", gcov_path.display()));
 
     let mut file = BufReader::new(&f);
+    if gcov_path.extension().unwrap() == "gz" {
```

Comment:
Let's split the function in two, one for `.json.gz` and one for `gcov`, then check the extension and call the right function, and return an error if the extension is unexpected